mqtt-iwatch-wot/README.md

```diff
@@ -0,0 +1,228 @@
+# Eclipse-Ditto-MQTT-iWatch
+This example presents how to configure Ditto to be able update things via MQTT. In this example we will create a iWatch from a WoT TM (Web of Things Thing Model). Our Digital Twin it will be updated via MQTT, using synthetic data.
+
+# Requirements
+1. Clone Ditto: ```git clone https://github.com/eclipse-ditto/ditto.git```
+
+2. Pull Mosquitto: ```docker pull eclipse-mosquitto```
+
+3. Clone Eclipse-Ditto-MQTT-iWatch: ```git clone https://github.com/bernar0507/Eclipse-Ditto-MQTT-iWatch.git```
+
+# Start Ditto and Mosquitto
+
+### Ditto: 
+```
+cd ditto
+```
+
+```
+git checkout tags/3.0.0
+```
+
+```
+cd ditto/deployment/docker
+```
+
+```
+docker compose up -d
+```
+
+### Mosquitto: 
+```
+docker run -it --name mosquitto --network docker_default -p 1883:1883 -v $(pwd)/mosquitto:/mosquitto/ eclipse-mosquitto
+```
+
+# Create the Policy
+```
+curl -X PUT 'http://localhost:8080/api/2/policies/org.Iotp2c:policy' -u 'ditto:ditto' -H 'Content-Type: application/json' -d '{
+    "entries": {
+        "owner": {
+            "subjects": {
+                "nginx:ditto": {
+                    "type": "nginx basic auth user"
+                }
+            },
+            "resources": {
+                "thing:/": {
+                    "grant": [
+                        "READ","WRITE"
+                    ],
+                    "revoke": []
+                },
+                "policy:/": {
+                    "grant": [
+                        "READ","WRITE"
+                    ],
+                    "revoke": []
+                },
+                "message:/": {
+                    "grant": [
+                        "READ","WRITE"
+                    ],
+                    "revoke": []
+                }
+            }
+        }
+    }
+}'
+```
+
+
+# Create the Thing
+We will use a WoT TM to create our Digital Twin:
+```
+curl --location --request PUT -u ditto:ditto 'http://localhost:8080/api/2/things/org.Iotp2c:iwatch' \
+  --header 'Content-Type: application/json' \
+  --data-raw '{
+      "policyId": "org.Iotp2c:policy",
+      "definition": "https://raw.githubusercontent.com/bernar0507/Eclipse-Ditto-MQTT-iWatch/main/iwatch/wot/iwatch.tm.jsonld"
+  }'
+```
+
+# Create a MQTT Connection
+We need to get the Mosquitto Ip adress from the container running mosquitto. 
+For that we need to use this to get the container id:
+```
+docker ps
+```
+
+After that we can get into the shell of the container:
+```
+docker exec -it CONTAINER_ID sh
+```
+
+After we get inside the container's shell we use the following command to retrieve the `IP_ADRESS` needed for when we create the connection, and for the `send_data_iwatch.py`:
+```
+ip address show
+```
+The `IP_ADRESS` is listed under interface "eth0" with the "inet" label and subnet mask.
```

Comment:
That would be a lot easier if you would just add the "mosquitto" to the docker-compose file of Ditto.
Then you could just use e.g. "mosquitto" as hostname (Docker's DNS resolves to the name you give it in the docker-compose.yml).

Reply:
I updated this today. I came up to the same conclusion as you. It's way easier the way you said.